app/questionnaire/questionnaire_schema.py

```diff
@@ -254,7 +254,7 @@ def get_list_name(self, current_location):
             current_location.section_id
         )
         if repeating_list and current_location.list_item_id:
-            list_name = repeating_list
+            return repeating_list
         return list_name
```

Comment:
Think this final return can be removed now as the method will return None anyway when the if condition isn't met.